src/langsmith/prompt-engineering-quickstart.mdx

```diff
@@ -3,103 +3,241 @@ title: Prompt engineering quickstart
+1. Add the following code to a `create_prompt` file, which:
 
-        <CodeGroup>
+    <CodeGroup>
 
     ```python Python
     from langsmith import Client
     from langchain_core.prompts import ChatPromptTemplate
 
-    # Connect to the LangSmith client
     client = Client()
 
-    # Define the prompt
     prompt = ChatPromptTemplate([
         ("system", "You are a helpful chatbot."),
         ("user", "{question}"),
     ])
 
-    # Push the prompt
-    client.push_prompt("my-prompt", object=prompt)
+    client.push_prompt("prompt-quickstart", object=prompt)
     ```
 
     ```typescript TypeScript
     import { Client } from "langsmith";
     import { ChatPromptTemplate } from "@langchain/core/prompts";
 
-    // Connect to the LangSmith client
     const client = new Client();
 
-    // Define the prompt
     const prompt = ChatPromptTemplate.fromMessages([
-        ["system", "You are a helpful chatbot."],
-        ["user", "{question}"]
+    ["system", "You are a helpful chatbot."],
+    ["user", "{question}"],
     ]);
 
-    // Push the prompt
-    await client.pushPrompt("my-prompt", {
-        object: prompt
+    await client.pushPrompt("prompt-quickstart", {
+    object: prompt,
     });
     ```
 
-        </CodeGroup>
+    </CodeGroup>
 
-    ### 3. Test a prompt
+    This creates an ordered list of messages, wraps them in `ChatPromptTemplate`, and then pushes the prompt by name to your [workspace](/langsmith/administration-overview#workspaces) for versioning and reuse.
 
-    To test a prompt, you need to pull the prompt, invoke it with the input values you want to test and then call the model with those input values. your LLM or application expects.
+1. Run `create_prompt`:
 
-        <CodeGroup>
+    <CodeGroup>
+
+    ```python Python
+    python create_prompt.py
+    ```
+
+    ```typescript TypeScript
+    npx tsx create_prompt.ts
+    ```
+
+    </CodeGroup>
+
+You can view your prompts with the [SDK](https://docs.smith.langchain.com/reference/python/reference) or the [UI](https://smith.langchain.com). With the SDK, call [`list_prompts`](https://docs.smith.langchain.com/reference/python/client/langsmith.client.Client#langsmith.client.Client.list_prompts) (Python) or [`listPrompts`](https://docs.smith.langchain.com/reference/js/classes/client.Client#listprompts) (TypeScript) to display the prompts in your [workspace](/langsmith/administration-overview#workspaces).
+
+## 3. Test a prompt
```

Comment:
I would combine step 3 and 4. 3 is a little anti-climactic because you're just loading the prompt. So flow could be
1. you’ll pull the prompt you created in step 2
2. create a new version of a prompt - edit the system message